lib/modules/platform/scm-manager/scm-client.ts

```diff
@@ -0,0 +1,156 @@
+import { Http } from '../../../util/http';
+import type {
+  HttpResponse,
+  InternalHttpOptions,
+} from '../../../util/http/types';
+import { resolveBaseUrl } from '../../../util/url';
+import type {
+  Link,
+  Page,
+  PullRequest,
+  PullRequestCreateParams,
+  PullRequestPage,
+  PullRequestUpdateParams,
+  Repo,
+  RepoPage,
+  ScmmHttpOptions,
+  User,
+} from './types';
+
+const URLS = {
+  ME: 'me',
+  ALL_REPOS: 'repositories?pageSize=1000000',
+  REPO: (repoPath: string) => `repositories/${repoPath}`,
+  PULLREQUESTS: (repoPath: string) => `pull-requests/${repoPath}`,
+  PULLREQUESTS_WITH_PAGINATION: (repoPath: string) =>
+    `pull-requests/${repoPath}?status=ALL&pageSize=1000000`,
+  PULLREQUEST_BY_ID: (repoPath: string, id: number) =>
+    `pull-requests/${repoPath}/${id}`,
+};
+
+const CONTENT_TYPES = {
+  ME: 'application/vnd.scmm-me+json;v=2',
+  REPOSITORY: 'application/vnd.scmm-repository+json;v=2',
+  REPOSITORIES: 'application/vnd.scmm-repositoryCollection+json;v=2',
+  GIT_CONFIG: 'application/vnd.scmm-gitDefaultBranch+json;v=2',
+  PULLREQUEST: 'application/vnd.scmm-pullRequest+json;v=2',
+  PULLREQUESTS: 'application/vnd.scmm-pullRequestCollection+json;v=2',
+};
+
+export default class ScmClient extends Http<ScmmHttpOptions> {
```

Comment:
Don't extend from Http here, add a new class to `utils/http/scm-manager.ts`.
Ideally do it as separate PR.